plugins/runner/internal/runtimehandler/binary/handler.go

```diff
@@ -0,0 +1,233 @@
+// Copyright © 2024 Cisco Systems, Inc. and its affiliates.
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build linux
+
+package binary
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"net"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"sync"
+	"syscall"
+
+	multierror "github.com/hashicorp/go-multierror"
+	"github.com/openclarity/vmclarity/plugins/runner/internal/runtimehandler"
+	"github.com/openclarity/vmclarity/plugins/runner/types"
+	"github.com/openclarity/vmclarity/plugins/sdk-go/plugin"
+	"github.com/openclarity/vmclarity/scanner/utils/containerrootfs"
+)
+
+type binaryRuntimeHandler struct {
+	config types.PluginConfig
+
+	cmd        *exec.Cmd
+	stdoutPipe io.ReadCloser
+	stderrPipe io.ReadCloser
+
+	pluginDir            string
+	inputDirMountPoint   string
+	pluginServerEndpoint string
+	ready                bool
+
+	mu sync.Mutex
+}
+
+func New(ctx context.Context, config types.PluginConfig) (runtimehandler.PluginRuntimeHandler, error) {
+	return &binaryRuntimeHandler{
+		config: config,
+	}, nil
+}
+
+func (h *binaryRuntimeHandler) Start(ctx context.Context) error {
+	h.mu.Lock()
+	defer h.mu.Unlock()
+
+	image, cleanup, err := containerrootfs.GetImageWithCleanup(ctx, h.config.ImageName)
+	if err != nil {
+		return fmt.Errorf("unable to get image(%s): %w", h.config.ImageName, err)
+	}
+	defer cleanup()
```

Comment:
We shouldn't do the cleanup here (currently it recursively deletes the host filesystem).

This cleanup runs at the end of the `Start` function when there is still an existing mount from the host's filesystem. Before we delete anything we should be able to ensure that there is no mount from the host fs.